test/authorize-hook.test.js

```diff
@@ -0,0 +1,52 @@
+var expect = require('chai').expect;
+var express = require('express');
+var RemoteObjects = require('../');
+var User = require('./e2e/fixtures/user');
+var fmt = require('util').format;
+var User = require('./e2e/fixtures/user');
+
+describe('authorization hook', function() {
+  var server;
+  var remotes;
+
+  before(function setupServer(done) {
+    var app = express();
+    remotes = RemoteObjects.create();
+    remotes.exports.User = User;
+    app.use(remotes.handler('rest'));
+    server = app.listen(0, '127.0.0.1', done);
+  });
+
+  after(function teardownServer(done) {
+    server.close(done);
+  });
+
+  describe('given a remotes object with an authorization hook', function() {
+    it('should be called when a remote method is invoked', function(done) {
+      var called = false;
+      remotes.authorization = function(ctx, next) {
+        called = true;
+        next();
+      };
+
+      invokeRemote(server.address().port,
+        function(err, session) {
+          expect(err).to.not.exist();
+          expect(session.userId).to.equal(123);
+          expect(called).to.equal(true);
+          done();
+        }
+      );
+    });
+  });
```

Comment:
Used your callStack array trick. Works like a charm. Though the confusing part is that since we are using `remotes.connect()` and `remotes.invoke()` the local before hook is also included in the `callStack`.